Example/Example Swift/MultipageDocumentService.swift

```diff
@@ -15,16 +15,28 @@ final class MultipageDocumentsService: DocumentServiceProtocol {
     var partialDocuments: [String: PartialDocumentInfo] = [:]
     var compositeDocument: GINIDocument?
     var analysisCancellationToken: BFCancellationTokenSource?
-    
+    var pendingAnalysisHandler: AnalysisCompletion?
```

Comment:
I am really against this. It just makes me think it's just a patch to make things somehow work. The Document service should know how to manage a document's lifecycle. Keep track if it's uploaded, if it makes sense to poll for extractions, etc. The only input it needs (from the UI) is if anyone is still interested in the result. And even in this case, I'd say it's better to follow the straight case and provide a way for the UI to say that it's no longer interested in the result and cancel.

In my experience, holding on to such closures because something might get delayed brings more problems than solutions.

Reply:
#241